### PR TITLE
Correct documentation regarding how to restore no `feature_states`

### PR DESCRIPTION
**Pull request by AthenaEryma**
This commit corrects the snapshot creation and restoration docs to
describe the usage of `"none"` to restore no feature states. Previously,
they incorrectly stated that using an empty array would accomplish this,
but specifying an empty array results in the default behavior (rather
than preventing feature state snapshot/restoration).

### CODE REVIEW

File: docs/reference/snapshot-restore/apis/create-snapshot-api.asciidoc [hunk 1]
```diff
@@ -133,7 +133,7 @@ default. If `include_global_state` is `false`, the snapshot includes no feature
 states by default.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/apis/create-snapshot-api.asciidoc:133]: I think we should explicitly state that an empty array results in default behavior here.

Many users may assume an empty array == no feature states.

File: docs/reference/snapshot-restore/apis/restore-snapshot-api.asciidoc [hunk 2]
```diff
@@ -183,7 +183,8 @@ state then the restore request will fail.
 If `include_global_state` is `true`, the request restores all feature states
 in the snapshot by default. If `include_global_state` is `false`, the request
 restores no feature states by default. To restore no feature states, regardless
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/apis/restore-snapshot-api.asciidoc:185]: Again, I'd explicitly state that an empty array results in default behavior here.

**Comment by gwbrown** [comment]
Thanks for the review @jrodewig, I've made explicit notes where you requested.

File: docs/reference/snapshot-restore/apis/create-snapshot-api.asciidoc [hunk 3]
```diff
@@ -132,8 +132,9 @@ If `include_global_state` is `true`, the snapshot includes all feature states by
 default. If `include_global_state` is `false`, the snapshot includes no feature
 states by default.
 +
-To exclude all feature states, regardless of the `include_global_state` value,
-specify an empty array (`[]`) or `none`.
+Note that specifying an empty array will result in the default behavior.To
```
Comment (rjernst) [inline docs/reference/snapshot-restore/apis/create-snapshot-api.asciidoc:135]: nit: space after period

**Review by rjernst — APPROVED** [review]
LGTM

**Review by jrodewig — APPROVED** [review]
LGTM with Ryan's suggestion.

**Pull request merged**